core/direct-rpc-server/src/response_channel.rs

```diff
@@ -15,19 +15,12 @@
 
 */
 
-use crate::api::SidechainApi;
-use itc_direct_rpc_server::{
-	rpc_connection_registry::ConnectionRegistry, rpc_responder::RpcResponder,
-};
-use itc_tls_websocket_server::connection::TungsteniteWsConnection;
-use itp_top_pool::basic_pool::BasicPool;
-use itp_types::Block;
+use crate::DirectRpcError;
+use std::string::String;
 
-type Hash = sp_core::H256;
+/// Response / status update channel for an RPC call.
+pub trait ResponseChannel<Token>: Send + Sync {
+	type Error: Into<DirectRpcError>;
```

Comment:
The `ResponseChannel` is now web-socket agnostic. This allowed me to further decouple the RPC server part from the web-sockets.